coast/TRANSECT.py

```diff
@@ -307,5 +308,118 @@ def plot_depth_integrated_transport(self, time, plot_info: dict, smoothing_windo
+                shape_ds = ( self.data_T.t_dim.size, z_levels.size, 
+                                self.data_T.r_dim.size )
+                sal = self.data_T.vosaline.to_masked_array()
+                temp = self.data_T.temperature.to_masked_array()                
+            except AttributeError:
+                shape_ds = ( 1, z_levels.size, self.data_T.r_dim.size )
+                sal = self.data_T.vosaline.to_masked_array()[np.newaxis,...]
+                temp = self.data_T.temperature.to_masked_array()[np.newaxis,...]
+            
+            sal_z_levels = np.ma.zeros( shape_ds )
+            temp_z_levels = np.ma.zeros( shape_ds )
+            density_z_levels = np.ma.zeros( shape_ds )
+            
+            s_levels = self.data_T.depth_0.to_masked_array()
+            lat = self.data_T.latitude.values
+            lon = self.data_T.longitude.values
+    
+            for it in np.arange(0, shape_ds[0]):
+                for ir in self.data_T.r_dim:
+                    if np.all(np.isnan(sal[it,:,ir])):
+                        density_z_levels[it,:,ir] = np.nan
+                        density_z_levels[it,:,ir].mask = True
+                    else:                      
+                        sal_func = interpolate.interp1d( s_levels[:,ir], sal[it,:,ir], 
+                                    bounds_error=False, kind='linear')
+                        temp_func = interpolate.interp1d( s_levels[:,ir], temp[it,:,ir], 
+                                    bounds_error=False, kind='linear')
+                        
+                        sal_z_levels[it,:,ir] = sal_func(z_levels.values)
+                        temp_z_levels[it,:,ir] = temp_func(z_levels.values)
+                        
+                
+            
+            pressure_absolute = np.ma.masked_invalid(
+                gsw.p_from_z( -z_levels.values[:,np.newaxis], lat ) ) # depth must be negative           
+                       
+            sal_absolute = np.ma.masked_invalid(
+                gsw.SA_from_SP( sal_z_levels, pressure_absolute, lon, lat ) )
+            sal_absolute = np.ma.masked_less(sal_absolute,0)
+
+            temp_conservative = np.ma.masked_invalid(
+                gsw.CT_from_pt( sal_absolute, temp_z_levels ) )
+    
+            density_z_levels = np.ma.masked_invalid( gsw.rho( 
+                sal_absolute, temp_conservative, pressure_absolute ) )
+            
+            coords={'depth_z_levels': (('z_dim'), z_levels.values),
+                    'latitude': (('r_dim'), self.data_T.latitude.values),
+                    'longitude': (('r_dim'), self.data_T.longitude.values)}
+            dims=['z_dim', 'r_dim']
+            
+            if shape_ds[0] != 1:
+                coords['time'] = (('t_dim'), self.data_T.time.values)
+                dims.insert(0, 't_dim')
+    
+            self.data_T['density_z_levels'] = xr.DataArray( np.squeeze(density_z_levels), 
+                    coords=coords, dims=dims )
+
```

Comment:
# Add attributes
            self.data_T['density_z_levels'].attrs = {
                                       'units': 'kg/m^3',
                                       'standard_name': name_str}

Not sure what the standard name ought to be for this though.